content/blog/pulumi-convert-terraform-improvements/index.md

```diff
@@ -0,0 +1,352 @@
+- other golang project artifacts (go.mod, go.sum)
+
+### Go specific cleanup
+
+The go code generator outputs everything in the Terraform code, even if it is
+unused. This is an error in a go program, so I had to manually remove
+unreferenced variables like GCP project, region, and planetscale service token.
+
+These are part of the output because in it's current iteration the code
+converter will convert everything, even if ultimately it is provider
+configuration (see [cleanup](#cleanup)) and not actual code.
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
+### Cleanup
+
+If we simply run `pulumi preview` this project unfortunately won't run right away.  
+We need to create our stack configuration that contains the GCP and Planetscale credentials.
+
+These were part of our `main.tf` file, but the code generator doesn't
+distinguish between provider config and code, so we need to handle this
+part on our own (for now).
+
+These can go in either your `Pulumi.yaml` or your stack configuration.  Places
+for them are already generated, you just need to fill them in:
+
+```yaml
+name: terraform-convert-example
+runtime: nodejs
+config:
+  google:project:
+    value: 'TODO: var.gcp_project' # fill in here
+  google:region:
+    value: 'TODO: var.gcp_region'
+  google:zone:
+    value: 'TODO: var.gcp_zone'
+  planetscale:serviceToken:
+    value: 'TODO: var.planetscale_service_token'
+  planetscale:serviceTokenName:
+    value: planetscaletoken
+```
+
+Also notice the configuration names differ from the pulumi counterparts "google" should be "gcp".
+
+After filling things in the `Pulumi.yaml` should look something like this:
+
+```yaml
+name: terraform-convert-example
+runtime: nodejs
+config:
+  gcp:project:
+    value: 'brandonpollack23'
+  gcp:region:
+    value: 'us-central1'
+  gcp:zone:
+    value: 'us-central1-a'
+  planetscale:serviceToken:
+    value: 'supersecrettoken'
+  planetscale:serviceTokenName:
+    value: planetscaletoken
+```
+
+and if you run `pulumi preview` it successfully generates a plan!
+
+There is still more you can do, the generated code could be cleaned up some as there
+are some unused variables, etc.
+
+## Limitations
+
+* We still have a bit of [unimplemented Terraform functionality](https://github.com/pulumi/pulumi-converter-terraform/issues/65) that we're tracking and are evaluating how to move forward.  For now when these functions are detected it will require some manual intervention on the converted project before you're ready to deploy.
+* If you define your terraform module in a parent directory of your deployment code, you'll encounter a [known bug](https://github.com/pulumi/pulumi-converter-terraform/issues/194), but a simple workaround is to restructure your Terraform code before running a conversion.
+* Terraform programs are dynamically typed, when converting to a type safe language sometimes a type is unknown and still needs to be added manually (as in the typescript example above).
```

Comment:
"dynamically typed, when" -> "dynamically typed. When" (grammar, break up run-on sentence)

Reply:
"type safe" -> "type-safe" (hyphenate to use as adjective)

Reply:
"still needs to" -> "needs to" (simplify grammar)

Reply:
"typescript" -> "TypeScript" (capitalize proper nouns)